File: sast-go/cases/accuracy/context_sensitive/argument_return_value_passing/argument_passing_value_012_T/argument_passing_value_012_T.go
// evaluation information start
// real case = true
// evaluation item = 准确度->上下文敏感分析->参数/返回值传递
// scene introduction = 
// level = 2
// bind_url = accuracy/context_sensitive/argument_return_value_passing/argument_passing_value_012_T/argument_passing_value_012_T
// evaluation information end

package argument_passing_value_012_T

func argument_passing_value_012_T(__taint_src string) {
	process(__taint_src)
}

func process(arg string) {
	__taint_sink(arg)
}

func __taint_sink(o interface{}) {
}

File: sast-python2/case/completeness/single_app_tracing/function_call/argument_passing/argument_passing_value_012_T.py
# -*- coding: utf-8 -*-
# evaluation information start
# real case = true
# evaluation item = 完整度->单应用跟踪完整度->函数和方法调用->参数传递
# scene introduction = 值传递嵌套函数
# level = 2
# bind_url = completeness/single_app_tracing/function_call/argument_passing/argument_passing_value_012_T
# evaluation information end
import os


def argument_passing_value_012_T(taint_src):
    def outer(input):
        def inner(inner_input):
            taint_sink(inner_input)

        inner(input)

    outer(taint_src)


def taint_sink(o):
    os.system(o)


if __name__ == u'__main__':
    taint_src = u"taint_src_value"
    argument_passing_value_012_T(taint_src)

